src/components/Staking/ETHLPStaking/TokenStakeInfo/index.js

```diff
@@ -0,0 +1,253 @@
+      getERC20Balance(tokenStakeInfo.stakingToken, selectedAccount);
+    }
+  };
+  const selectTokenStakeAddressInfo = (account) => {
+    if (!selectedAccount
+      || !tokenStakeAddressInfo[account]
+      || tokenStakeAddressInfo[account].loading
+      || tokenStakeAddressInfo[account].error
+    ) {
+      return undefined;
+    }
+    return tokenStakeAddressInfo[account];
+  };
+
+  const erc20FromSelector = (erc20Address) => {
+    const mapped = erc20Info[erc20Address];
+    if (!mapped || mapped.loading || mapped.error) {
+      return undefined;
+    }
+    return mapped;
+  };
+  const erc20BalanceFromSelector = (erc20Address, account) => {
+    const mapped = account && erc20Balance[erc20Address]?.[account];
+    if (!mapped || mapped.loading || mapped.error) {
+      return undefined;
+    }
+    return mapped;
+  };
+
+  React.useEffect(() => {
+    getTokenStake();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  React.useEffect(() => {
+    getAddressRelated();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [selectedAccount, tokenStakeInfo]);
+
+  React.useEffect(() => {
+    if (tokenStakeInfo && !tokenStakeInfo.error) {
+      getERC20Info(tokenStakeInfo.rewardToken);
+      getERC20Info(tokenStakeInfo.stakingToken);
+    }
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [tokenStakeInfo]);
+
+  if (!tokenStakeInfo || tokenStakeInfoLoading) {
+    return <div>Loading...</div>;
+  }
+
+  if (tokenStakeInfo.error) {
+    return (
+      <div className={styles.error}>
+        Something went wrong
+      </div>
+    );
+  }
+
+  const rewardTokenInfo = erc20FromSelector(tokenStakeInfo?.rewardToken);
+  const stakingTokenInfo = erc20FromSelector(tokenStakeInfo?.stakingToken);
+  const rewardTokenBalance = erc20BalanceFromSelector(tokenStakeInfo?.rewardToken, selectedAccount);
+  const stakingTokenBalance = erc20BalanceFromSelector(tokenStakeInfo?.stakingToken, selectedAccount);
+  const addressInfo = selectTokenStakeAddressInfo(selectedAccount);
+  const tokensStakedLoader = () => (addressInfo
+    ? `${formatCustom(addressInfo.stake[0].toString(), tokenStakeInfo.stakingTokenDecimals)}${
+      stakingTokenInfo ? ` ${stakingTokenInfo.symbol}` : ''}`
+    : 'Loading...');
+  const rewardsLoader = () => (addressInfo
+    ? `${formatCustom(addressInfo.stake[1].toString(), tokenStakeInfo.rewardTokenDecimals)}${
+      rewardTokenInfo ? ` ${rewardTokenInfo.symbol}` : ''}`
+    : 'Loading...');
+  const rewardInAccountLoader = () => (!rewardTokenBalance || !tokenStakeInfo
+    ? 'Loading...'
+    : `${formatCustom(rewardTokenBalance.balance.toString(), tokenStakeInfo.rewardTokenDecimals)}${
+      rewardTokenInfo ? ` ${rewardTokenInfo.symbol}` : ''}`);
+  const stakingInAccountLoader = () => (!stakingTokenBalance || !tokenStakeInfo
+    ? 'Loading...'
+    : `${formatCustom(stakingTokenBalance.balance.toString(), tokenStakeInfo.stakingTokenDecimals)}${
```

Comment:
Heh, thats a bit of overengineering for the task, but fine :P

Reply:
Linter begged to differ :-)